Doc/library/tempfile.rst

```diff
@@ -66,6 +66,11 @@ The module defines the following user-callable items:
 
       The :py:data:`os.O_TMPFILE` flag is now used if available.
 
+   .. versionchanged:: 3.7
```

Comment:
So this isn't entirely true since you didn't need to change anything for support, so this really has been supported since Python 3.6.